runtime/compiler/runtime/SymbolValidationManager.cpp

```diff
@@ -69,14 +69,17 @@ TR::SymbolValidationManager::SymbolValidationManager(TR::Region &region, TR_Reso
      _seenSymbolsSet((SeenSymbolsComparator()), _region)
    {
    TR::Compilation* comp = TR::comp();
-   J9VMThread *vmThread = comp->j9VMThread();
-   TR_J9VM *fej9 = (TR_J9VM *)TR_J9VMBase::get(vmThread->javaVM->jitConfig, vmThread);
-
+   TR_J9VM *fej9 = comp->fej9vm();
```

Comment:
There's a reason the SVM used `TR_J9VM *fej9 = (TR_J9VM *)TR_J9VMBase::get(vmThread->javaVM->jitConfig, vmThread);` instead of `TR_J9VM *fej9 = comp->fej9vm();`.

The front end from the comp is the `TR_J9SharedCacheVM`; this will result in problems when calling `getComponentClassFromArrayClass` below because it will invoke the SVM (which is still being constructed). In general, the SVM uses the `TR_J9VM` object specifically because the query it's making is not something that requires a validation (because the SVM is performing the validation).

Reply:
I see. That means when we start generating AOT at the server, we will have to do an if statement to get the server VM, every time `TR_J9VMBase::get` is called on the server.

Reply:
Well I think when AOT is generated at the server, you're going to have to use the base JaaS VM class that gets the information from the client. We'll have to think about how to structure the `TR_J9SharedCacheVM` and a potential `TR_JaaSSharedCacheVM` to prevent duplicated code.